vraptor-core/src/main/java/br/com/caelum/vraptor/http/iogi/IogiParametersProvider.java

```diff
@@ -90,11 +90,12 @@ private Object instantiateOrAddError(Parameters parameters, List<Message> errors
 	private List<Target<Object>> createTargets(ControllerMethod method) {
 		Method javaMethod = method.getMethod();
 		List<Target<Object>> targets = new ArrayList<>();
+		TypeExtractor typeExtractor = new TypeExtractor(method.getController().getType());
```

Comment:
This is really needed? This method is used in one place only. So I'd keep
as private method in the same class, to using kiss principle.
On Jul 6, 2015 15:39, "Rodrigo Turini" <notifications@github.com> wrote:

> In
> vraptor-core/src/main/java/br/com/caelum/vraptor/http/iogi/IogiParametersProvider.java
> <https://github.com/caelum/vraptor4/pull/992#discussion_r33967540>:
>
> > @@ -90,11 +90,12 @@ private Object instantiateOrAddError(Parameters parameters, List<Message> errors
> >  	private List<Target<Object>> createTargets(ControllerMethod method) {
> >  		Method javaMethod = method.getMethod();
> >  		List<Target<Object>> targets = new ArrayList<>();
> > +		TypeExtractor typeExtractor = new TypeExtractor(method.getController().getType());
>
> what about changing TypeExtractor to be a CDI managed bean?
> so it could be specialized -- extensible
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/caelum/vraptor4/pull/992/files#r33967540>.
>